### PR TITLE
Feature/unittest/highlightservice

### PR DESCRIPTION
**Pull request by RamonPeek**
## Description

Test-coverage for the HighlightService methods.

## Type of change

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] This change requires a documentation update

## Checklist

-   [ ] My code follows the style guidelines of this project
-   [ ] I did not update API Controllers, if I did, I added/changed Postman tests
-   [ ] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have made corresponding changes to the documentation
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [ ] New and existing unit tests pass locally with my changes
-   [ ] I updated the changelog with an end-user readable description
-   [ ] I assigned this pull request to the correct project board to update the sprint board

## Steps to Test or Reproduce

Outline the steps to test or reproduce the PR here.
These steps will be used during release testing.

1. Right click on the HighlightServiceTest.cs class
2. Select 'Run tests'

### CODE REVIEW

**Comment by jcoenen96** [comment]
Can you merge this branch with develop into this branch? @RamonPeek

File: 2_Services.Tests/HighlightServiceTest.cs [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+            Assert.DoesNotThrow(() => {
+                RepositoryMock.Verify(repository => repository.GetHighlightsAsync(), Times.Once);
+            });
+
+            Assert.AreEqual(highlights, retrievedHighlights);
+            Assert.AreEqual(10, retrievedHighlights.Count);
+        }
+
+        [Test]
+        public async Task GetHighlightsAsync_NoHighlights()
```
Comment (jcoenen96) [inline 2_Services.Tests/HighlightServiceTest.cs:45]: No commentary

File: 2_Services.Tests/HighlightServiceTest.cs [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    public class HighlightServiceTest : ServiceTest<Highlight, HighlightService, IHighlightRepository>
+    {
+        protected new IHighlightService Service => (IHighlightService) base.Service;
+
+        /// <summary>
+        /// Test whether the repository method is called and no changes have been applied to the object
+        /// </summary>
+        /// <returns></returns>
+        [Test]
+        public async Task GetHighlightsAsync_GoodFlow([HighlightDataSource(10)] List<Highlight> highlights)
```
Comment (jcoenen96) [inline 2_Services.Tests/HighlightServiceTest.cs:25]: Missing the null test,
Missing what happens if projects is null

File: 3_Repositories.Tests/DataGenerators/HighlightDataGenerator.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+namespace Repositories.Tests.DataGenerators
+{
+    /// <summary>
+    /// FakeDataGenerator for the projects
+    /// </summary>
+    public class HighlightDataGenerator : FakeDataGenerator<Highlight>
+    {
+        /// <summary>
+        /// Initializes the projectDataGenerator
```
[the remark this answers is not in the document]
Comment (RamonPeek) [reply 3_Repositories.Tests/DataGenerators/HighlightDataGenerator.cs:14]: This is set in the baseclass as far as I understand. Can't seem to find where to change this at the moment

[the remark this answers is not in the document]
Comment (jcoenen96) [reply 3_Repositories.Tests/DataGenerators/HighlightDataGenerator.cs:14]: Your comments says: "Initializes the projectDataGenerator."
This should be "Initializes the highlightDataGenerator."

File: 2_Services.Tests/HighlightServiceTest.cs (hunk 2 above)
Comment (jcoenen96) [reply 2_Services.Tests/HighlightServiceTest.cs:25]: There is no test that tests what happens if project is null

File: 3_Repositories.Tests/DataGenerators/HighlightDataGenerator.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
+using Bogus;
+using Models;
+using Repositories.Tests.DataGenerators.Base;
+using System;
+
+namespace Repositories.Tests.DataGenerators
+{
+    /// <summary>
+    /// FakeDataGenerator for the projects
```
[the remark this answers is not in the document]
Comment (jcoenen96) [reply 3_Repositories.Tests/DataGenerators/HighlightDataGenerator.cs:9]: FakeDataGenerator for the projects -> FakeDataGenerator for the highLights

File: 2_Services.Tests/HighlightServiceTest.cs [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
```
Comment (StijnGroenen) [inline 2_Services.Tests/HighlightServiceTest.cs:25]: Missing parameter documentation

File: 2_Services.Tests/HighlightServiceTest.cs [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+
+        }
+
+
+        /// <summary>
+        /// Test whether the repository method is called if no project is provided and no changes have been applied to the object assuming no highlights were provided
+        /// </summary>
+        /// <returns></returns>
+        [Test]
+        public async Task GetHighlightsAsync_HighlightWithoutProject([HighlightDataSource] Highlight highlight)
```
Comment (StijnGroenen) [inline 2_Services.Tests/HighlightServiceTest.cs:75]: Missing parameter documentation

File: 3_Repositories.Tests/DataSources/HighlightDataSourceAttribute.cs [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+        public HighlightDataSourceAttribute()
+        {
+            fakeDataGenerator = new HighlightDataGenerator();
+        }
+
+        /// <summary>
+        /// Initializes highlightDataSourceAttribute
+        /// and setting the amount of highlights to be generated
+        /// </summary>
+        public HighlightDataSourceAttribute(int amount) : this()
```
Comment (StijnGroenen) [inline 3_Repositories.Tests/DataSources/HighlightDataSourceAttribute.cs:34]: Missing parameter documentation

**Comment by StijnGroenen** [comment]
Please add this PR to the changelog.

**Comment by Brend-Smits** [comment]
Please merge Develop into this branch for the folder fix.
Make sure that the unit tests you have added are in the new folder structure now.

**Pull request merged**